MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/TextFieldFormattingActivity.java

```diff
@@ -0,0 +1,149 @@
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textSize;
+
+/**
+ * Use runtime styling to adjust the font, size, and color of SymbolLayer text fields.
+ */
+public class TextFieldFormattingActivity extends AppCompatActivity {
+
+  private final String[] textFonts = new String[] {"Roboto Black", "Arial Unicode MS Regular"};
+  private final Float[] textSizes = new Float[] {25f, 13f};
+  private final String[] textColors = new String[] {"#00FF08", "#ffd43a"};
+  private MapView mapView;
+  private boolean fontChanged = false;
+  private boolean sizeChanged = false;
+  private boolean colorChanged = false;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_dds_text_field_formatting);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(@NonNull MapboxMap mapboxMap) {
+        mapboxMap.setStyle(Style.DARK, new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull final Style style) {
+
+            Toast.makeText(TextFieldFormattingActivity.this, getString(R.string.instruction_toast),
+              Toast.LENGTH_SHORT).show();
+
+            // Set click listeners for the text adjustment buttons
+            FloatingActionButton textFontFab = findViewById(R.id.fab_toggle_font);
+            textFontFab.setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                adjustText(style, textFont(fontChanged ? new String[] {textFonts[1]} : new String[] {textFonts[0]}));
+                fontChanged = !fontChanged;
+              }
+            });
+
+            FloatingActionButton textSizeFab = findViewById(R.id.fab_toggle_text_size);
+            textSizeFab.setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                adjustText(style, textSize(sizeChanged ? textSizes[1] : textSizes[0]));
+                sizeChanged = !sizeChanged;
+              }
+            });
+
+            FloatingActionButton textColorFab = findViewById(R.id.fab_toggle_text_color);
+            textColorFab.setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                adjustText(style, textColor(colorChanged ? textColors[1] : textColors[0]));
+                colorChanged = !colorChanged;
+              }
+            });
```

Comment:
can we refactor above statements, they contain similar functionality.
Additionally assigning and setting the click listener can be done in one line instead.